src/range-request.ts

```diff
@@ -1,10 +1,48 @@
+import { Config } from ".";
 import parseMultipartBody from "./multipart";
 import { LengthIntegrityError } from "./resolver";
 
+async function resolveIndividualPromises(
+  url: string,
+  ranges: { start: number; end: number; expectedLength?: number }[],
+) {
+  console.log("resolving ranges individually");
+  // fallback to resolving ranges individually
+  const individualRangePromises = ranges.map(
+    async ({ start, end, expectedLength }) => {
+      const rangeHeader = `${start}-${end}`;
+      const res = await fetch(url, {
+        headers: { Range: `bytes=${rangeHeader}` },
+      });
+
+      const totalLength = Number(
+        res.headers.get("Content-Range")!.split("/")[1],
+      );
+      if (expectedLength && totalLength !== expectedLength) {
+        throw new LengthIntegrityError();
+      }
+      return {
+        data: await res.arrayBuffer(),
+        totalLength: totalLength,
+      };
+    },
+  );
+  return await Promise.all(individualRangePromises);
+}
+
 export async function requestRanges(
   url: string,
   ranges: { start: number; end: number; expectedLength?: number }[],
+  config: Config,
 ): Promise<{ data: ArrayBuffer; totalLength: number }[]> {
+  const { useMultipartByteRanges } = config;
+  if (
+    useMultipartByteRanges === false ||
+    useMultipartByteRanges === undefined
```

Comment:
```suggestion
    useMultipartByteRanges === false
```

The default behavior should be true